src/api/server/authn.rst

```diff
@@ -345,3 +345,124 @@ headers to CouchDB with related requests:
 
 Note that you don't need to request :ref:`session <api/auth/session>`
 to be authenticated by this method if all required HTTP headers are provided.
+
+.. _api/auth/jwt:
+
+JWT Authentication
+====================
+
+.. note::
+    To use this authentication method make sure that the
+    ``{chttpd_auth, jwt_authentication_handler}`` value is added to the
+    list of the active :config:option:`chttpd/authentication_handlers`:
+
+    .. code-block:: ini
+
+        [chttpd]
+        authentication_handlers = {chttpd_auth, cookie_authentication_handler}, {chttpd_auth, jwt_authentication_handler}, {chttpd_auth, default_authentication_handler}
+
+``JWT authentication`` enables CouchDB to use externally generated JWT tokens
+instead of defining users or roles in the ``_users`` database.
+
+The JWT authentication handler requires that all JWT tokens are signed by a key that
+CouchDB has been configured to trust (there is no support for JWT's "NONE" algorithm).
+
+Additionally, CouchDB can be configured to reject JWT tokens that are missing a
+configurable set of claims (e.g, a CouchDB administrator could insist on the ``exp`` claim).
+
+All claims presented in a JWT token are validated if presented, regardless of whether they
+are required.
+
+Two sections of config exist to configure JWT authentication;
+
+.. code-block:: ini
+
+    ;[jwt_auth]
+    ; List of claims to validate
+    ; required_claims =
+
+The `required_claims` config setting is a comma-separated list of additional mandatory
+JWT claims that must be present in any presented JWT token. A `:code 400:Bad Request` is sent
+if any are missing.
+
+The ``alg`` claim is mandatory as it used to lookup the correct key for verifying the
+signature.
+
+The ``sub`` claim is mandatory and is used as the CouchDB user's name if the JWT token
+is valid.
+
+A private claim called ``_couchdb.roles`` is optional. If presented, it is a JSON
+array of strings, as is used as the CouchDB user's roles list if the JWT token
+is valid.
```

Comment:
Wording of the second sentence seems slightly awkward. Would this work?

```
A private claim called ``_couchdb.roles`` is optional. If present and valid, it is a JSON
array of strings, as is used as the CouchDB user's roles.
```

Reply:
should be "and is used as the" in the original comment. does that make more sense?